Chessington.GameEngine/Square.cs

```diff
@@ -16,6 +16,23 @@ public static Square At(int row, int col)
             return new Square(row, col);
         }
 
+        public bool IsInBounds()
+        {
+            var row = Row;
+            if (row < 0 || row > 7)
```

Comment:
you can use Row instead of row for these checks. It will get translated to Row.get(), which is row